### ISSUE TITLE
Avoid global variables

### ISSUE THREAD

**Issue by ateucher** [bcgov/bcgroundwater#1]
Refactor dplyr and ggplot code using standard evaluation "escape hatches"

**Comment by stephhazlitt** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Update package with indicator update (2018)](https://trello.com/c/KmNwT2Z2/8-update-package-with-indicator-update-2018)

**Comment by steffilazerte** [comment]
So far I've addressed this by:
- using `aes_string()` in ggplot2 function calls
- using `.data$variable` notation in tidyverse function calls (i.e. `mutate`, `group_by`, etc)

Using `.data$variable` avoids the possibility of accidentally using a variable in the global environment but it still results in one undeclared global variable: `.data`

The only other way (that I've found) to avoid global variables entirely is to use the `rlang` package, for example:

```
g %>% 
  dplyr::group_by(rlang::UQS(rlang::parse_expr("Well_Num")))
```

This seems overly cumbersome and requires another package. It's entirely possible there's an easier way, but I haven't been able to figure it out, as of yet.

I would suggest either:

- Using the `globalVariables()` function to declare all NSE variables we use (removes Notes from the output of check)
- Use `.data` and `aes_string` and be satisfied with one undeclared global variable (`.data`) [Currently implemented]

**Comment by ateucher** [comment]
That's great! Yeah, I think the `rlang` way is the new way to do things, but I think since these in general are not user-supplied variables (they are column names that _must_ be in the data, correct?) the `.data` approach seems entirely reasonable. I also noticed [here](https://github.com/tidyverse/dplyr/blob/master/vignettes/programming.Rmd#different-data-sets) that you can use `@importFrom rlang .data` to avoid .data being identified as a global variable. (And `dplyr` already imports `rlang`, so I'm not too worried about adding it to our `Imports:`.

**Comment by steffilazerte** [comment]
Fixed: 8787364
